src/_main.yml

```diff
@@ -86,8 +86,11 @@ repository:
       "2":
         patterns:
           - name: variable.other.enummember.hcl
-            comment: Block label
-            match: '[\"\-\w]+'
+            comment: Block label (String Literal)
+            match: '\"[^\"\r\n]*\"'
```

Comment:
```suggestion
            match: '\"[^\"\r\n]+\"'
```

Reply:
So in this case I would argue that 'zero or more' is correct for syntax highlighting. `""` is a valid string literal (from what I can tell in the language spec) but it _may_ be invalid block label. Depends on the decoder I imagine.

If we don't tag it as block label, it'll highlight the same as the block itself which will look wrong. This would also be a change in behaviour from previously.

Note - I should probably add a test for that!

Current scoping;
![image](https://github.com/hashicorp/syntax/assets/5819622/1b252575-8846-4d54-bae4-35d16ce2bb89)

Reply:
Oh I see, totally fair point! We have already made the decision elsewhere in the grammar to not try to make the it 100% correct. For one it's impossible with regex-based model and secondly it's not always beneficial to the user anyway at this level (early syntax highlighting) as "lack of highlighting" is a very poor communication channel for syntax errors.

TL;DR IMO the validity of empty label can be a question of further validation in LS and the grammar can just treat it as valid.

I do agree though that we should add a test for empty label to cement this decision.

Reply:
Test added.